crates/rspack_plugin_javascript/src/dependency/new_url/mod.rs

```diff
@@ -0,0 +1,145 @@
+use swc_core::ecma::utils::{member_expr, quote_ident, quote_str};
+use swc_core::ecma::{ast::*, atoms::JsWord};
+
+#[derive(Debug, Eq, Clone)]
+pub struct NewUrlDependency {
+  id: Option<DependencyId>,
+  parent_module_identifier: Option<ModuleIdentifier>,
+  request: JsWord,
+  span: Option<ErrorSpan>,
+  #[allow(unused)]
+  ast_path: JsAstPath,
+  unresolved_mark: Mark,
+}
+
+// Do not edit this, as it is used to uniquely identify the dependency.
+impl PartialEq for NewUrlDependency {
+  fn eq(&self, other: &Self) -> bool {
+    self.parent_module_identifier == other.parent_module_identifier && self.request == other.request
+  }
+}
+
+// Do not edit this, as it is used to uniquely identify the dependency.
+impl std::hash::Hash for NewUrlDependency {
+  fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+    self.parent_module_identifier.hash(state);
+    self.request.hash(state);
+    self.category().hash(state);
+    self.dependency_type().hash(state);
+  }
+}
+
+impl NewUrlDependency {
+  pub fn new(
+    request: JsWord,
+    span: Option<ErrorSpan>,
+    ast_path: JsAstPath,
+    unresolved_ctxt: SyntaxContext,
+  ) -> Self {
+    Self {
+      id: None,
+      parent_module_identifier: None,
+      request,
+      span,
+      ast_path,
+      unresolved_mark: unresolved_ctxt.outer(),
+    }
+  }
+}
+
+impl Dependency for NewUrlDependency {
+  fn id(&self) -> Option<DependencyId> {
+    self.id
+  }
+  fn set_id(&mut self, id: Option<DependencyId>) {
+    self.id = id;
+  }
+  fn parent_module_identifier(&self) -> Option<&ModuleIdentifier> {
+    self.parent_module_identifier.as_ref()
+  }
+
+  fn set_parent_module_identifier(&mut self, module_identifier: Option<ModuleIdentifier>) {
+    self.parent_module_identifier = module_identifier;
+  }
+
+  fn category(&self) -> &DependencyCategory {
+    &DependencyCategory::CommonJS
+  }
+
+  fn dependency_type(&self) -> &DependencyType {
+    &DependencyType::CjsRequire
+  }
+}
+
+impl ModuleDependency for NewUrlDependency {
+  fn request(&self) -> &str {
+    &self.request
+  }
+
+  fn user_request(&self) -> &str {
+    &self.request
+  }
+
+  fn span(&self) -> Option<&ErrorSpan> {
+    self.span.as_ref()
+  }
+}
+
+impl CodeGeneratable for NewUrlDependency {
+  fn generate(
+    &self,
+    code_generatable_context: &mut CodeGeneratableContext,
+  ) -> rspack_error::Result<CodeGeneratableResult> {
+    let CodeGeneratableContext { compilation, .. } = code_generatable_context;
+    let mut code_gen = CodeGeneratableResult::default();
+
+    let require_span = DUMMY_SP.apply_mark(self.unresolved_mark);
+
+    if let Some(id) = self.id() {
+      if let Some(module_id) = compilation
+        .module_graph
+        .module_graph_module_by_dependency_id(&id)
+        .map(|m| m.id(&compilation.chunk_graph).to_string())
+      {
+        code_gen.visitors.push(
+          create_javascript_visitor!(&self.ast_path, visit_mut_expr(n: &mut Expr) {
```

Comment:
`create_javascript_visitor!(exact &self.ast_path, visit_mut_new_expr(n: &mut NewExpr)` did not work.

I have given up on trying to further investigate the research after an hour of debug.

Reply:
@h-a-n-a

Reply:
@magic-akari Hi, could you add some detail about the case of *not work*? I don't quite understand.

Reply:
> @magic-akari Hi, could you add some detail about the case of _not work_? I don't quite understand.

@h-a-n-a  replace this code block with following code:
```JavaScript
create_javascript_visitor!(exact &self.ast_path, visit_mut_new_expr(n: &mut NewExpr) {
        n.args = Some(vec![]);
}),
```

run the test case.
Expected result: The `URL` parameters will be cleared.
Actual result: Test failed withoute specific detail.

Reply:
@magic-akari You need to tweak the code here, which intercepts the visitor. https://github.com/web-infra-dev/rspack/blob/62bad724be07c0d7ab30251e1a52ddd7100f687d/crates/rspack_plugin_javascript/src/visitors/dependency/code_generation.rs#L260.